samples/switchless/README.md

```diff
@@ -45,33 +46,38 @@ A host worker thread checks and retrieves `job` from the shared memory region. I
 runtime to process the `job` by unmarshaling the parameters, then dispatching to the callee function, and finally
 relaying the result back to the trusted Open Enclave runtime, which is further forwarded back to the caller.
 
+Switchless ECALLs work in a similar manner. The `job` is created by the untrusted Open Enclave runtime (host-side) and
+posted to a shared memory region. An enclave workter thread checks and retrieves the `job` from the shared region
+and processes the `job` by unmarshaling the parameters, then dispatching to the callee function, and finally
+relaying the result back to the untrusted Open Enclave runtime, which is further forwarcded back to the caller.
+
 If an enclave supports multiple simultaneous ECALLs, multiple simultaneous switchless OCALLs could be made from the
 enclave. We use multi-threaded host workers in that scenario. Open Enclave
-allows users to configure how many host worker threads are to be created for servicing switchless OCALLs. The
-following example illustrates how to do that. A word of caution is that too many host worker threads might increase
+allows users to configure how many worker threads are to be created for servicing switchless OCALLs and ECALLs. The
+following example illustrates how to do that. A word of caution is that too many worker threads might increase
 competition of cores between threads and degrade the performance. Therefore, if a enclave has switchless calls
-enabled, Open Enclave caps the number of host worker threads for it to the number of enclave threads specified.
+enabled, Open Enclave caps the number of worker threads for it to the number of enclave threads specified.
 
-With the current implementation, we recommend that users avoid using more host worker threads than the minimum of:
+With the current implementation, we recommend that users avoid using more worker threads than the minimum of:
 
-1. the number of simultaneously active enclave threads, and
-2. the number of cores that are potentially available to host worker threads.
+1. the number of simultaneously active caller threads, and
+2. the number of cores that are potentially available to worker threads.
```

Comment:
Is there a platform-agnostic API that an app can use to get the number of cores on the machine?

Reply:
I am not aware of such an API. The closest is C++11 [std::thread::hardware_concurrency](https://en.cppreference.com/w/cpp/thread/thread/hardware_concurrency). However, it is only a hint.